resources/RequestResetPassword.py

```diff
@@ -0,0 +1,46 @@
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask_restful import Resource
+from flask import request
+from middleware.user_queries import user_check_email
+from middleware.reset_token_queries import add_reset_token
+import os
+import uuid
+import requests
+
+
+class RequestResetPassword(Resource):
+    def __init__(self, **kwargs):
+        self.psycopg2_connection = kwargs["psycopg2_connection"]
+
+    def get(self):
+        try:
+            data = request.get_json()
+            email = data.get("email")
+            cursor = self.psycopg2_connection.cursor()
+            user_data = user_check_email(cursor, email)
+            id = user_data["id"]
+            token = uuid.uuid4().hex
+            add_reset_token(cursor, email, token)
+            self.psycopg2_connection.commit()
+
+            body = f"To reset your password, click the following link: {os.getenv('VITE_VUE_APP_BASE_URL')}/reset-password/{token}"
```

Comment:
Oh, one other thing here. We want the base url of the client. This is sending the base url of the API.

Reply:
This will require an additional env variable VITE_VUE_API_BASE_URL which will have to replace all current instances of VITE_VUE_APP_BASE_URL in the client code. They will be the same locally but will need to be updated in DO when these are merged